src/devices/ant_antplus.c

```diff
@@ -0,0 +1,198 @@
+    if (bitbuffer->bits_per_row[0] < 137 || bitbuffer->bits_per_row[0] > 160) {
+        return DECODE_ABORT_LENGTH;
+    }
+
+    // find a data package and extract data buffer
+    bit_offset = bitbuffer_search(bitbuffer, 0, 0, preamble, sizeof(preamble) * 8) + sizeof(preamble) * 8;
+    if (bit_offset + sizeof(b1) * 8 > bitbuffer->bits_per_row[0]) { // did not find a big enough package
+        return DECODE_ABORT_LENGTH;
+    }
+
+    // ANT and ANT+ packes have either aa or 55 preamble, depending on the first bit of
+    // the following byte. The best way to know which is being used, is to rely on the CRC
+    bitbuffer_extract_bytes(bitbuffer, 0, bit_offset, b1, sizeof(b1) * 8);     // if preamble is AA
+    bitbuffer_extract_bytes(bitbuffer, 0, bit_offset + 1, b2, sizeof(b2) * 8); // if preamble is 55, read starting 1 bit further right
+
+    // calculate CRC for both alternatives (Schrodinger's CRC in superposition state)
+    uint16_t crc1 = crc16ccitt(b1, 15);
+    uint16_t crc2 = crc16ccitt(b2, 15);
+
+    // all variables initialized to avoid bogus [-Wmaybe-uninitialized] warning
+    // byte[6] value not decoded/displayed, it's always = 10
+    uint16_t net_key    = 0;
+    uint16_t id         = 0;
+    uint8_t device_type = 0;
+    uint8_t tx_type     = 0;
+    char payload[8 * 3 + 1]; // payload is 8 hex pairs for ANT and ANT+
+
+    // use CRC to decide if the data packet is using aa or 55 as preamble. Bytes 15 and 16 are the CRC
+    // (peek into Schrodinger's box and collapse the CRC wavefunction)
+    if (((b1[15] << 8) | b1[16]) == crc1) { // valid preamble is aa
+        net_key     = (b1[1] << 8) | b1[0];
+        id          = (b1[3] << 8) | b1[2];
+        device_type = b1[4];
+        tx_type     = b1[5];
+        sprintf(payload, "%02x %02x %02x %02x %02x %02x %02x %02x", b1[7], b1[8], b1[9], b1[10], b1[11], b1[12], b1[13], b1[14]);
+
+        // display ANT or ANT+ depending on the network key used. ANT+ only uses aa preamble
+        char model[5] = "ANT\0";
+        if (0xc5a6 == net_key)
+            strncpy(model, "ANT+\0", 5);
+
+        /* clang-format off */
+        data = data_make(
+                "model",       "",               DATA_STRING, model,
+                "net_key",     "Net key",        DATA_FORMAT, "0x%04x", DATA_INT, net_key,
+                "device_num",  "Device #",       DATA_FORMAT, "0x%04x", DATA_INT, id,
+                "device_type", "Device type",    DATA_INT,    device_type,
+                "tx_type",     "TX type",        DATA_INT,    tx_type,
+                "payload",     "Payload",        DATA_STRING, payload,
+                "CRC",         "CRC",            DATA_FORMAT, "0x%04x", DATA_INT, crc1,
+                NULL);
+        /* clang-format on */
+
+        decoder_output_data(decoder, data);
+        return 1;
+    }
+    else if (((b2[15] << 8) | b2[16]) == crc2) { // valid preamble is 55
```

Comment:
With restructured code logic this code section will not be needed.

Reply:
Yep